packages/react-reconciler/src/ReactFiberScope.js

```diff
@@ -31,7 +31,9 @@ function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
 }
```

Comment:
I think you might need to check for `fiber.memoizedState.dehydrated === null` here.

https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberSuspenseComponent.js#L37

If dehydrated is non-null it's a not-yet hydrating boundary.

Reply:
Addressed your feedback. Thank you :)